CurrencyExchangeServiceWCF/CurrencyExchangeServiceWCF/Service.svc.cs

```diff
@@ -1,81 +1,100 @@
 
-            if (!GetCurrencyCodes().Contains(toCurrency))
-            {
-                return "Currency not available in the CurrencyConverter currencies list.";
+                if (!isCurrencyValid || !isValueValid)
+                {
+                    r.ReturnValue = "Input cannot be NULL!";
+                    return r;
+                }
 
-            }
+                double ConversionRateDouble;
+                string ConversionRateString;
+            
+                if (toCurrency.Equals("USD"))
+                {
+                    r.ReturnValue = value;
+                    return r;
+                }
+                if (toCurrency.Equals("EUR"))
+                    toCurrency = "USD";
 
-            double conversionRateDouble;
-            string conversionRateString;
-            double valueDouble;
+                if (!GetCurrencyCodes().Contains(toCurrency))
+                {
+                    soapException = new SoapException("Currency not available in the CurrencyConverter currencies list.", SoapException.ClientFaultCode, "");
+                    throw soapException;
 
-            try
-            {
-                conversionRateString = GetActualConversionRate(toCurrency);
-                conversionRateDouble = Convert.ToDouble(conversionRateString);
-                valueDouble = Convert.ToDouble(value);
+                }
+                if (String.IsNullOrEmpty(value))
+                {
+                    soapException = new SoapException("Wert darf nicht 0 sein.", SoapException.ClientFaultCode, "");
+                    throw soapException;
+                }
+           
+                ConversionRateString = GetActualConversionRate(toCurrency);
+                ConversionRateDouble = Convert.ToDouble(ConversionRateString);
+           
+                if (toCurrency.Equals("USD")) 
+                {
+                    value = (Convert.ToDouble(value) / ConversionRateDouble).ToString();
+                }
+                else
+                {
+                    value = (Convert.ToDouble(value) * ConversionRateDouble).ToString();
+                }           
+                r.ReturnValue = value;         
+                return r;
             }
             catch (Exception)
             {
-                return "Input value has invalid number format.";
-            }
-            if (toCurrency.Equals("USD"))
-            {
-                value = (valueDouble / conversionRateDouble).ToString();
-            }
-            else
-            {
-                value = (valueDouble * conversionRateDouble).ToString();
+                soapException = new SoapException("Wert konnte nicht umgerechnet werden.", SoapException.ClientFaultCode, "");
+                throw soapException;
             }
-
-            return value;
-
         }
 
-        private string GetActualConversionRate(string currency)
+        private string GetActualConversionRate(string Currency)
```

Comment:
This changes is against the naming convention. Parameters should start with a lower case letter.

Reply:
fixed